### PR TITLE
Add network (live/testnet) switcher to transfer widget

### PR DESCRIPTION
**Pull request by johny**
Allow switching between mainnet & testnet networks for ecosystem transfer widget:

Screenshot 2023-05-25 at 19 01 32

Note, when switching between mainnet&testnet the wallet connection will reset.

Also ended up with plenty of additional improvements:
- Better display of loading state in the UI
- Validate withdrawal amount vs balance
- Also withdaw funds stuck in the intermediary account
- Show links to edgeware.app when block explorer is not available for test network

### CODE REVIEW

**Review by ShankarWarang — DISMISSED** [review]

**Comment by ShankarWarang** [comment]
Is it possible to add auto network switching when user clicks on the mainnet/testnet or when connects EVM wallet? And if the EdgeEVM/BereEVM is not configured, will it prompt the EVM wallet (metamask) to add the corresponding network?

**Comment by johny** [comment]
@ShankarWarang - the logic should be fixed now ;-)

**Review by ShankarWarang — APPROVED** [review]

**Comment by ShankarWarang** [comment]
I'm getting 'Insufficient balance' error now while depositing from EdgeWASM to EdgeEVM. @johny

**Comment by johny** [comment]
> I'm getting 'Insufficient balance' error now while depositing from EdgeWASM to EdgeEVM. @johny

Try again! I've fixed some issues related to that.

File: lib/crypto/withdrawal.ts [hunk 1]
```diff
@@ -48,17 +57,37 @@ export const processEVMWithdrawal = async (
   }
 
   // Request withdrawal
+  const BUFFER = 1;
   const amount = Number(inputAmount);
+  const withdrawExtra = availableBalance - BUFFER > amount;
+  const actualAmount = withdrawExtra ? availableBalance - BUFFER : amount;
+  const difference = Math.floor(actualAmount - BUFFER - amount);
```
Comment (ShankarWarang) [inline lib/crypto/withdrawal.ts:64]: No need to deduct 'BUFFER'.

**Review by ShankarWarang — CHANGES_REQUESTED** [review]
Requesting the minor change for lib/crypto/withdrawal.ts : L64

**Comment by johny** [comment]
> Requesting the minor change for lib/crypto/withdrawal.ts : L64

Adressed in [2eaf08b](https://github.com/edgeware-network/Edgeware-Website/pull/70/commits/2eaf08b523922facd9d8058ab5c99e1ba74a36a6)

**Review by ShankarWarang — APPROVED** [review]

**Pull request merged**